azkeys/keyvault.go

```diff
@@ -191,6 +200,9 @@ func (kv *KeyVault) Sign(
 	keyName := GetKeyName(keyID)
 	keyVersion := GetKeyVersion(keyID)
 
+	span, ctx := tracing.StartSpanFromContext(ctx, "KeyVault Sign")
```

Comment:
do we need to finish this span before the verify span starts?

Reply:
same question for other spans, it looks like `KeyVault GetKeyVersions` is the only span that calls span.Finish() as well as deferring it.

What happens to `span.Finish()` if we defer it and then also call it before the function finishes?

Reply:
Sorry, I don't understand this comment.  The span started on line 203 will end at the end of the function (due to the deferred Finish()) where does the verify span start?

Reply:
the verify span starts on line 286.

I am trying to understand if the sign span will run from line 203 -> the end of the function (because of defer).
If it does, does it not encroach on the time taken in the verify span?

Is that something we desire to have?

Reply:
The other comment is on line 106 we call `span.Finish()` as well as deferring it at line 98. I just wanted to understand if there are any repercussions of finishing a span before the end of the function when it also gets called because of the defer.

Reply:
span.Finish() is defined in the type to be idempotent so that should be fine.

I think you may be confused about the spans in Sign and Verify, they do not overlap